test/aoc.test.js

```diff
@@ -106,6 +106,36 @@ Game 5: 6 red, 1 blue, 3 green; 2 blue, 1 red, 2 green`
   expect(res).toBe(8)
 })
 
+test("day2-part2", () => {
+  let input = `Game 1: 3 blue, 4 red; 1 red, 2 green, 6 blue; 2 green
+Game 2: 1 blue, 2 green; 3 green, 4 blue, 1 red; 1 green, 1 blue
+Game 3: 8 green, 6 blue, 20 red; 5 blue, 4 red, 13 green; 5 green, 1 red
+Game 4: 1 green, 3 red, 6 blue; 3 green, 6 red; 3 green, 15 blue, 14 red
+Game 5: 6 red, 1 blue, 3 green; 2 blue, 1 red, 2 green`
+
+  let udf = udf_stdlib
+
+  let line = rh`.input | udf.split "\\n" | .*line
+                       | udf.split ":"`
+ 
+  let cube = rh`${line}.1 | udf.split ";" | .*hand
+                          | udf.split "," | .*group
+                          | udf.split " "`
+  
+  let num   = rh`${cube}.1 | udf.toNum`
+  let color = rh`${cube}.2`
+
+  let lineRes = {
+    "-": api.keyval(color, api.max(num))
+  }
+
```

Comment:
Could this be written as `max num | group ${color}`?

Reply:
This is giving incorrect result. If I do this I will get something like `{",3,blue,,4,red,,1,red,,2,green,,6,blue,,2,green,2": 6}` for each line. After looking at the AST after parsing I think the problem is that `group ${color}` will incorrectly make the first parameter of `keyval`.
The AST looks like this:
```
{
  "_IGNORE_": {
    xxkey: "keyval",
    xxparam: {
      0: { Array [2] }, // Incorrect. This should be ${color} which is a "get" operation
      1: { xxkey: "max", xxpath: { ... } } // This is max ${num}
    }
  }
}
```

If I manually modify the AST I will get correct result:
```
let lineRes = rh`max ${num} | (group ${color})`
lineRes._IGNORE_.xxparam[0] = {
  xxpath: "get",
  xxparam: lineRes._IGNORE_.xxparam[0]
}
```

This might be a bug when parsing something like `group ${color}`.

Reply:
What's the expansion of the "Array [2]" above? (I think in VSCode you can expand output through the UI, or in the terminal you can get it to print more detail with `console.dir(res, {depth: 20})`)

Reply:
`color` is a get operation `let color = rh"${cube}.2"` so it should be parsed into
```
{
  xxpath: "get",
  xxparam: {...}
}
```
The `Array [2]` in the AST is what this xxparam should be. So it works when I manually put it into xxparam of the get operation.

Reply:
I will look at those functions

Reply:
Line 406 in parser.js is causing this.
In function transPrimitiveApply():
```
} else if (p == "group" && args.length < 2) {
      // partial application -- this will later turn into a keyval object
      return { xxpath: "group", xxparam: args[0].xxparam } 
}
```
should be
```
} else if (p == "group" && args.length < 2) {
      // partial application -- this will later turn into a keyval object
      return { xxpath: "group", xxparam: args[0] } 
}
```

I'll change that and push again